infrastructure/src/main/java/org/corfudb/infrastructure/LogUnitServer.java

```diff
@@ -535,6 +535,7 @@ public void shutdown() {
         executor.shutdown();
         logCleaner.shutdown();
         batchWriter.close();
+        streamLog.close();
```

Comment:
This seems like it was also leaking file descriptors, since `ConcurrentMap<String, SegmentHandle> writeChannels` wasn't being cleaned up properly.

Reply:
yep, probably it would be nice to have a test that checks that 'close' methods are executed for all classes. 
This line of code was removed some time ago (looks like it happened accidentally) and we reintroduced file descriptor leaks again